pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/operator/V2Operator.java

```diff
@@ -0,0 +1,51 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.query.runtime.operator;
+
+import java.util.List;
+import org.apache.pinot.core.common.Operator;
+import org.apache.pinot.core.operator.BaseOperator;
+import org.apache.pinot.query.runtime.blocks.TransferableBlock;
+import org.slf4j.LoggerFactory;
+
+
+public abstract class V2Operator extends BaseOperator<TransferableBlock> implements AutoCloseable {
```

Comment:
(nit) My suggestion is to refrain from v1 and v2 terminology (which I know we all do a lot in discussions) because the current engine isn't going away.

Probably `MultiStageOperator` is a better name for the abstraction.

Reply:
Or may be `IntermediaryStageOperator` / `MultiStageEngineOperator` .....

Reply:
Good point. Changed the name to MultiStageOperator